### DISCUSSION TITLE
LAB3 spec Puzzle

### DISCUSSION THREAD

**Discussion by kevin861222** [bol-edu/HLS-SOC-Discussions#118] [Q&A]
請問lab3 中 fir 發生axi_reset_n的負緣觸發時，需要reset兩個bram的資料嗎？
通常asic中的bram沒有reset port，這樣是不是意味著要花上11或12個cycle才能完成reset呢？

**Comment by ZheChen-Bill** [comment]
以目前我對bram的瞭解，如果要對bram進行reset，會需要自己幫他們做reset，因此會需要花多個cycle，原因也正如你所說bram沒有reset port。

**Comment by boledulab** [comment]
Yes, the state machine should have a state, say "INIT" to initialize the internal states, including the SRAM content, status, ...
